core/dblookupext/historyRepository.go

```diff
@@ -126,25 +136,15 @@ func (hr *historyRepository) recordMiniblock(blockHeaderHash []byte, blockHeader
 		DestinationShardID: miniblock.GetReceiverShardID(),
 	}
 
-	// If we are on metachain and the miniblock is towards us, then we can also apply hyperblock coordinates at commit & record time,
+	// If we are on metachain and the miniblock is towards us, then we simulate a notarization notification at commit & record time,
 	// since there will be no notification via blockTracker.Register(*)NotarizedHeadersHandler anyway.
 	selfIsMeta := hr.selfShardID == core.MetachainShardId
 	receiverIsMeta := miniblock.GetReceiverShardID() == core.MetachainShardId
 	if selfIsMeta && receiverIsMeta {
-		miniblockMetadata.NotarizedAtSourceInMetaNonce = blockHeader.GetNonce()
-		miniblockMetadata.NotarizedAtSourceInMetaHash = blockHeaderHash
-		miniblockMetadata.NotarizedAtDestinationInMetaNonce = blockHeader.GetNonce()
-		miniblockMetadata.NotarizedAtDestinationInMetaHash = blockHeaderHash
-	}
-
-	// Here we need to use queued notifications
-	notification, ok := hr.notarizedAtSourceNotifications.Get(string(miniblockHash))
-	if ok {
-		notificationTyped := notification.(*notarizedAtSourceNotification)
-		miniblockMetadata.NotarizedAtSourceInMetaNonce = notificationTyped.metaNonce
-		miniblockMetadata.NotarizedAtSourceInMetaHash = notificationTyped.metaHash
-
-		hr.notarizedAtSourceNotifications.Remove(string(miniblockHash))
+		hr.pendingNotarizedAtBothNotifications.Set(string(miniblockHash), &notarizedNotification{
+			metaNonce: blockHeader.GetNonce(),
```

Comment:
blockHeader is always a meta block?

Reply:
Fixed as discussed internally (removed "simulated notification").